core/src/main/java/org/bitcoinj/wallet/DerivationPathFactory.java

```diff
@@ -0,0 +1,94 @@
+
+    public DerivationPathFactory(NetworkParameters params) {
+        this.params = params;
+        this.coinType = new ChildNumber(params.getCoinType(), true);
+    }
+
+    public ImmutableList<ChildNumber> blockchainIdentityRegistrationFundingDerivationPath() {
+        return ImmutableList.<ChildNumber>builder()
+                .add(FEATURE_PURPOSE)
+                .add(coinType)
+                .add(new ChildNumber(12, true))
+                .build();
+    }
+
+    public ImmutableList<ChildNumber> blockchainIdentityDerivationPath() {
+        return ImmutableList.<ChildNumber>builder()
+                .add(FEATURE_PURPOSE)
+                .add(coinType)
+                .add(ChildNumber.FIVE_HARDENED)
+                .add(ChildNumber.ZERO_HARDENED)
+                .add(ChildNumber.ZERO_HARDENED)
+                .build();
+    }
+
+    public ImmutableList<ChildNumber> masternodeHoldingsDerivationPath() {
+        return ImmutableList.<ChildNumber>builder()
+                .add(FEATURE_PURPOSE)
+                .add(coinType)
+                .add(new ChildNumber(3, true))
+                .add(ChildNumber.ZERO_HARDENED)
+                .build();
+    }
+
+    public ImmutableList<ChildNumber> masternodeVotingDerivationPath() {
+        return ImmutableList.<ChildNumber>builder()
+                .add(FEATURE_PURPOSE)
+                .add(coinType)
+                .add(new ChildNumber(3, true))
+                .add(ChildNumber.ONE_HARDENED)
+                .build();
+    }
+
+    public ImmutableList<ChildNumber> masternodeOwnerDerivationPath() {
+        return ImmutableList.<ChildNumber>builder()
+                .add(FEATURE_PURPOSE)
+                .add(coinType)
+                .add(new ChildNumber(3, true))
+                .add(new ChildNumber(2, true))
+                .build();
+    }
+
+    public ImmutableList<ChildNumber> masternodeOperatorDerivationPath() {
+        return ImmutableList.<ChildNumber>builder()
+                .add(FEATURE_PURPOSE)
+                .add(coinType)
+                .add(new ChildNumber(3, true))
+                .add(new ChildNumber(3, true))
+                .build();
+    }
+
+    public ImmutableList<ChildNumber> bip44DerivationPath(int account) {
+        return ImmutableList.<ChildNumber>builder()
+                .add(new ChildNumber(44, true))
+                .add(coinType)
+                .add(new ChildNumber(account, true))
+                .build();
+    }
+
+    public ImmutableList<ChildNumber> bip32DerivationPath(int account) {
+        return ImmutableList.<ChildNumber>builder()
+                .add(new ChildNumber(account, true))
+                .build();
+    }
+
+    // Do we need this?
+    private static DerivationPathFactory instance;
+    public static DerivationPathFactory get(NetworkParameters params) {
+        if(instance == null) {
+            instance = new DerivationPathFactory(params);
+        }
+        return instance;
+    }
```

Comment:
This is here to create a singleton of the DerivationPathFactory class.  Not sure if having a singleton is useful, especially in unit tests that may require derivation paths for two different networks.